types/react-native/index.d.ts

```diff
@@ -2942,7 +2942,17 @@ export interface DatePickerIOSProps extends ViewProps {
     /**
      * The currently selected date.
      */
-    date: Date;
+    date?: Date;
+    
+    /**
+     * Provides an initial value that will change when the user starts selecting
+     * a date. It is useful for simple use-cases where you do not want to deal
+     * with listening to events and updating the date prop to keep the
+     * controlled state in sync. The controlled state has known bugs which
+     * causes it to go out of sync with native. The initialDate prop is intended
+     * to allow you to have native be source of truth.
+     */
+    initialDate?: Date;
```

Comment:
As `?Date` in Flow means `Date | null | undefined`, we maybe should make it `initialDate?: Date | null` here and above for the date - wdyt? (see: https://flow.org/en/docs/types/maybe/)

In addition I would add a simple test, passing in the allowed types :)

Reply:
@ca057 then we would have to change all the other fields too, as they are typed as maybe as well. I oriented myself on how the current props were typed.

Reply:
I think it makes sense to add new typings correctly without necessarily applying retroactively to all existing ones.